internal/loader/aftermarket_device_loader.go

```diff
@@ -54,3 +63,35 @@ func (ad *AftermarketDeviceLoader) BatchGetLinkedAftermarketDeviceByVehicleID(ct
 
 	return results
 }
+
+// BatchGetLinkedAftermarketDeviceByID implements the dataloader for finding aftermarket devices by their ids and returns
+// them in the order requested
+func (ad *AftermarketDeviceLoader) BatchGetLinkedAftermarketDeviceByID(ctx context.Context, ids []int) []*dataloader.Result[*model.AftermarketDevice] {
+	results := make([]*dataloader.Result[*model.AftermarketDevice], len(ids))
+
+	devices, err := models.AftermarketDevices(models.AftermarketDeviceWhere.ID.IN(ids)).All(ctx, ad.db.DBS().Reader)
+	if err != nil {
+		for i := range ids {
+			results[i] = &dataloader.Result[*model.AftermarketDevice]{Data: nil, Error: err}
+		}
+		return results
+	}
+
+	amByVehicleID := map[int]*models.AftermarketDevice{}
```

Comment:
Why are there all these references to vehicle ids in this function? I'm confused.

I think we should build up a map from aftermarket device id to aftermarket device.

Reply:
I think the reason the tests aren't blowing up is that everything has id 1 so it's all kind of working out.